tests/50federation/33room-get-missing-events.pl

```diff
@@ -413,3 +411,125 @@ sub sytest_user_and_room_fixture {
+
+      $outbound_client->join_room(
+         server_name => $first_home_server,
+         room_id     => $room_id,
+         user_id     => $user_id,
+         version     => "6",
+      )->then( sub {
+         my ( $room ) = @_;
+
+         # TODO: We happen to know the latest event in the server should be my
+         #   m.room.member state event, but that's a bit fragile
+         my $latest_event = $room->get_current_state_event( "m.room.member", $user_id );
+
+         log_if_fail "Latest event", $latest_event;
+
+         # Generate but don't send an event
+         my $missing_event = $room->create_and_insert_event(
+            type => "m.room.message",
+
+            sender  => $user_id,
+            content => {
+               body    => "Message 1",
+               bad_val => 1.1,
+            },
+         );
+
+         log_if_fail "Missing event", $missing_event;
+
+         # Generate another one and do send it so it will refer to the
+         # previous in its prev_events field
+         my $sent_event = $room->create_and_insert_event(
+            type => "m.room.message",
+
+            # This would be done by $room->create_and_insert_event anyway but lets be
+            #   sure for this test
+            prev_events => $room->make_event_refs( $missing_event ),
+
+            sender  => $user_id,
+            content => {
+               body => "Message 2",
+            },
+         );
+         my $sent_event_id = $room->id_for_event( $sent_event );
+
+         log_if_fail "Sent event", $sent_event;
+
+         Future->needs_all(
+            $inbound_server->await_request_get_missing_events( $room_id )
+            ->then( sub {
+               my ( $req ) = @_;
+               my $body = $req->body_from_json;
+
+               log_if_fail "Body", $body;
+
+               assert_json_keys( $body, qw( earliest_events latest_events limit ));
+               # TODO: min_depth but I have no idea what it does
+
+               assert_json_list( my $earliest = $body->{earliest_events} );
+               @$earliest == 1 or
+                  die "Expected a single 'earliest_event' ID";
+               assert_eq( $earliest->[0], $room->id_for_event( $latest_event ),
+                  'earliest_events[0]' );
+
+               assert_json_list( my $latest = $body->{latest_events} );
+               @$latest == 1 or
+                  die "Expected a single 'latest_events' ID";
+               assert_eq( $latest->[0], $sent_event_id,
+                  'latest_events[0]' );
+
+               my @events = $datastore->get_backfill_events(
+                  start_at    => $latest,
+                  stop_before => $earliest,
+                  limit       => $body->{limit},
+               );
+
+               log_if_fail "Backfilling", @events;
+
+               $req->respond_json( {
+                  events => \@events,
+               } );
+
+               log_if_fail "Done here";
+
+               Future->done(1);
```

Comment:
should probably decide if we're using `Future->done;` or `Future->done(1);`. I'd vote for the former, but ymmv

Reply:
We seem very inconsistent on this throughout SyTest, but I'll make the new changes be `Future->done;`.